packages/use-contractkit/src/modals/connect.tsx

```diff
@@ -148,10 +149,14 @@ export const ConnectModal: React.FC<ConnectModalProps> = ({
           ([providerKey, provider]) =>
             typeof window !== 'undefined' &&
             provider.showInList() &&
-            Object.keys(allScreens).find((screen) => screen === providerKey)
+            Object.keys(allScreens)
+              .filter((screen) =>
+                screen.toLowerCase().includes(search.toLowerCase())
+              )
```

Comment:
what about cases where there is not a search yet?

Reply:
This is filtering (line 153) inside a filter (line 148) which is less than optimal. 

the is probably a way to do this without the n squared filtering

Reply:
I agree this is not optimal, but we're talking about tiny amount of data here, we don't have 100s of screens nor 100s of providers